TubeMQ consumers configure the wrong standby masterIP and set it to the first，consumers will throw exceptions.

TubeMQ consumers configure the wrong standby masterIP and set it to the first，consumers will throw exceptions. And then，consumers re-register successfully after 5min.

The reason for this problem is because consumers don‘t poll connection list if the connection times out.

[ERROR] [BaseMessageConsumer]  Register to master failed.
java.util.concurrent.TimeoutException
        at com.tencent.tubemq.corerpc.client.CallFuture.get(CallFuture.java:139)
        at com.tencent.tubemq.corerpc.netty.NettyClient.call(NettyClient.java:157)
        at com.tencent.tubemq.corerpc.RpcServiceFailoverInvoker.callMethod(RpcServiceFailoverInvoker.java:69)
        at com.tencent.tubemq.corerpc.AbstractServiceInvoker.invoke(AbstractServiceInvoker.java:57)
        at com.sun.proxy.$Proxy0.consumerRegisterC2M(Unknown Source)
        at com.tencent.tubemq.client.consumer.BaseMessageConsumer.startMasterAndBrokerThreads(BaseMessageConsumer.java:563)
        at com.tencent.tubemq.client.consumer.BaseMessageConsumer.completeSubscribe(BaseMessageConsumer.java:306)
        at com.tencent.tubemq.client.consumer.SimplePushMessageConsumer.completeSubscribe(SimplePushMessageConsumer.java:76)

--- a/tubemq-core/src/main/java/com/tencent/tubemq/corerpc/RpcServiceFailoverInvoker.java
+++ b/tubemq-core/src/main/java/com/tencent/tubemq/corerpc/RpcServiceFailoverInvoker.java
@@ -45,7 +45,7 @@ public class RpcServiceFailoverInvoker extends AbstractServiceInvoker {
         super(clientFactory, serviceClass, conf);
         this.masterInfo = masterInfo;
         this.masterNodeCnt = masterInfo.getNodeHostPortList().size();
-        getNextClient();
+        getNextClient(false);
     }
 
     @Override
@@ -53,9 +53,9 @@ public class RpcServiceFailoverInvoker extends AbstractServiceInvoker {
                              Object arg, Callback callback) throws Throwable {
         if (currentClient == null
                 || !currentClient.isReady()) {
-            getNextClient();
+            getNextClient(false);
         }
-        long currentCounter = retryCounter.get();
+        int currentCounter = retryCounter.get();
         RequestWrapper requestWrapper =
                 new RequestWrapper(PbEnDecoder.getServiceIdByServiceName(targetInterface),
                         RpcProtocol.RPC_PROTOCOL_VERSION,
@@ -66,37 +66,46 @@ public class RpcServiceFailoverInvoker extends AbstractServiceInvoker {
         Throwable t = null;
         for (int i = 0; i < masterNodeCnt; i++) {
             if (currentClient != null) {
-                ResponseWrapper responseWrapper =
-                        currentClient.call(requestWrapper, callback,
-                                requestTimeout, TimeUnit.MILLISECONDS);
-                if (responseWrapper != null) {
-                    if (responseWrapper.isSuccess()) {
-                        return responseWrapper.getResponseData();
-                    } else {
-                        Throwable remote =
-                                MixUtils.unwrapException(new StringBuilder(512)
-                                        .append(responseWrapper.getErrMsg()).append("#")
-                                        .append(responseWrapper.getStackTrace()).toString());
-                        if ((IOException.class.isAssignableFrom(remote.getClass()))
-                                || (StandbyException.class.isAssignableFrom(remote.getClass()))) {
-                            if (currentCounter == retryCounter.get()) {
-                                getNextClient();
-                                currentCounter++;
-                            }
-                            t = remote;
+                try {
+                    ResponseWrapper responseWrapper =
+                            currentClient.call(requestWrapper, callback,
+                                    requestTimeout, TimeUnit.MILLISECONDS);
+                    if (responseWrapper != null) {
+                        if (responseWrapper.isSuccess()) {
+                            return responseWrapper.getResponseData();
                         } else {
-                            throw remote;
+                            Throwable remote =
+                                    MixUtils.unwrapException(new StringBuilder(512)
+                                            .append(responseWrapper.getErrMsg()).append("#")
+                                            .append(responseWrapper.getStackTrace()).toString());
+                            if ((IOException.class.isAssignableFrom(remote.getClass()))
+                                    || (StandbyException.class.isAssignableFrom(remote.getClass()))) {
+                                if (currentCounter == retryCounter.get()) {
+                                    getNextClient(true);
+                                    currentCounter++;
+                                }
+                                t = remote;
+                            } else {
+                                throw remote;
+                            }
                         }
+                    } else {
+                        break;
                     }
-                } else {
-                    break;
+                } catch (Throwable e) {
+                    // If the call throws an exception and the master address is not polled, we need to try again.
+                    if (currentCounter == retryCounter.get()) {
+                        getNextClient(true);
+                        currentCounter++;
+                    }
+                    t = e;
                 }
             } else {
-                int index = (int) (currentCounter % (long) (masterNodeCnt));
+                int index = (currentCounter & Integer.MAX_VALUE) % masterNodeCnt;
                 t = new IOException(new StringBuilder(512).append("Connect server ")
                         .append(masterInfo.getNodeHostPortList().get(index)).append(" failure!").toString());
                 if (currentCounter == retryCounter.get()) {
-                    getNextClient();
+                    getNextClient(false);
                     currentCounter++;
                 }
             }
@@ -107,8 +116,15 @@ public class RpcServiceFailoverInvoker extends AbstractServiceInvoker {
         return null;
     }
 
-    private synchronized Client getNextClient() {
-        if (currentClient == null || !currentClient.isReady()) {
+    private synchronized Client getNextClient(boolean forceChange) {
+        // forceChange : force to create a new connection to the master
+        if (currentClient != null) {
+            if (forceChange || !currentClient.isReady()) {
+                currentClient.close();
+                currentClient = null;
+            }
+        }
+        if (currentClient == null) {
             Client client = null;
             int retryTimes = masterNodeCnt;
             List<String> addressList = masterInfo.getNodeHostPortList();

